mm2src/mm2_main/src/lp_swap/swap_watcher.rs

```diff
@@ -328,26 +366,29 @@ impl State for RefundTakerPayment {
     type Result = ();
 
     async fn on_changed(self: Box<Self>, watcher_ctx: &mut WatcherContext) -> StateResult<Self::Ctx, Self::Result> {
-        let locktime = watcher_ctx.data.taker_payment_lock;
-        loop {
-            match watcher_ctx
-                .taker_coin
-                .can_refund_htlc(locktime + WAIT_FOR_TAKER_REFUND)
-                .compat()
-                .await
-            {
-                Ok(CanRefundHtlc::CanRefundNow) => break,
-                Ok(CanRefundHtlc::HaveToWait(to_sleep)) => Timer::sleep(to_sleep as f64).await,
-                Err(e) => {
-                    error!("Error {} on can_refund_htlc, retrying in 30 seconds", e);
-                    Timer::sleep(30.).await;
-                },
+        if std::env::var("SKIP_WAIT_FOR_REFUND").is_err() {
+            loop {
+                match watcher_ctx
+                    .taker_coin
+                    .can_refund_htlc(
```

Comment:
`can_refund_htlc` expects an actual *payment lock time*. It might be error-prone to increase it. According to MTP requirements, this may additionally delay refund attempt.
I think we can simply sleep until `locktime + wait_for_refund` here and only then start checking `can_refund_htlc`.